src/sql/src/plan/statement/ddl.rs

```diff
@@ -1407,7 +1411,217 @@ pub fn plan_create_views(
                 materialize: materialized,
             }))
         }
-        CreateViewsDefinitions::Source { .. } => bail!("cannot create view from source"),
+        CreateViewsDefinitions::Source {
+            name: source_name,
+            targets,
+        } => {
+            let source_connector = normalize::unresolved_object_name(source_name.clone())
+                .map_err(anyhow::Error::new)
+                .and_then(|name| {
+                    scx.catalog
+                        .resolve_item(&name)
+                        .and_then(|item| item.source_connector())
+                        .map(|s| s.clone())
+                        .map_err(anyhow::Error::new)
+                });
+            match source_connector? {
+                SourceConnector::External {
+                    connector:
+                        ExternalSourceConnector::Postgres(PostgresSourceConnector {
+                            conn: _,
+                            publication: _,
+                            details,
+                            ..
+                        }),
+                    ..
+                } => {
+                    // If the user specified targets, validate they are all in the PostgresSourceDetails
+                    // otherwise create them from the contents of PostgresSourceDetails
```

Comment:
Why do we have to do this validation here since we're also validating the same thing down below? It looks like this whole section could be replaced by:

```rust
let targets = targets.unwrap_or_else(|| {
    details
        .tables
        .iter()
        .map(|t| {
            let name = UnresolvedObjectName::qualified(&[&t.namespace, &t.name]);
            CreateViewsSourceTarget {
                name: name.clone(),
                alias: Some(name),
            }
        })
        .collect()
});
```

Reply:
Seems reasonable, this section was done first before I realized that there was also the whole view definition and view planning loop below that used the same data.